### ISSUE TITLE
Disappearing background-image on `onEnter`

### ISSUE THREAD

**Issue by dnzk** [brigade/react-waypoint#212]
Hi, awesome plugin, it helps me a lot. I have issues however, involving the plugin and images set as background-images of divs. So I have a bunch of divs with set width and height on different positions in the page. Each of these divs has image as its background, some of them are wrapped with `Waypoint`, some of them are not. Given this set up, when I scroll down the page and trigger the `onEnter` event of whatever `Waypoint` element currently entering the page, it seems to rerender the whole page, causing the background image of the aforementioned divs to be gone for a noticeable split second, whether the div is wrapped with `Waypoint` or not. As comparison, I also have `img` elements in the page whose when `onEnter` happens don't seem to even blink. What is the best way to mitigate this background-image issue? Cheers!

**Comment by trotzig** [comment]
Hi @dnzk, thanks for using the component! 

Can you share a little bit more about your page? More specifically, I'd like to know what happens in the `onEnter` handlers. Are you updating state? 

If you can come up with a minimal repro case, this will likely be easier to resolve. Thanks!

**Comment by dnzk** [comment]
Hi @trotzig, I've fixed the issue by moving the `background-image` declaration to its own class (CSS class), in a parent component, which I think effectively put it outside the component life cycle. By the way, to give more context, the issue only happens in Safari 10.0.1 (can't confirm if it's doing it in other versions of Safari too), and I'm using [styled-components](https://github.com/styled-components/styled-components) for the styling which allows me to declare css in js kind of style inside the component I created. In some instances, I do update state on `onEnter` to cue the state based animation to happen, which I think triggers the component's rerender, along with all the style declaration in it (including the background-image and its source). So by moving the background-image to another class in a parent component that won't be rerendered by child component's rerendering, I think I have solved the issue. Thank you for the prompt reply!
